src/core/init/lifecycle.js

```diff
@@ -36,14 +36,27 @@ export function Lifecycle(Base) {
         if (index >= queue.length) {
           next(data);
         } else if (typeof hookFn === 'function') {
+          const errTitle = `Docsify plugin error (${hookName})`;
+
           if (hookFn.length === 2) {
-            hookFn(data, result => {
-              data = result;
-              step(index + 1);
-            });
+            try {
+              hookFn(data, result => {
+                data = result;
+              });
+            } catch (err) {
+              console.error(errTitle, err);
```

Comment:
Can we keep this in the separate PR?

Reply:
Yes. This change should not have been merged into this PR. Sorry about that. Good catch. I'll remove it.

Reply:
@trusktr -- Reverted. Thanks again for the catch.